pytorch_lightning/core/hooks.py

```diff
@@ -295,36 +295,23 @@ def on_after_backward(self):
 
         """
 
-    def backward(self, trainer, loss: Tensor, optimizer: Optimizer, optimizer_idx: int) -> None:
+    def backward(self, loss: Tensor, optimizer: Optimizer, *args, **kwargs) -> None:
         """
-        Override backward with your own implementation if you need to.
-
-        Args:
-            trainer: Pointer to the trainer
-            loss: Loss is already scaled by accumulated grads
-            optimizer: Current optimizer being used
-            optimizer_idx: Index of the current optimizer being used
-
-        Called to perform backward step.
-        Feel free to override as needed.
-
-        The loss passed in has already been scaled for accumulated gradients if requested.
+        Call this directly from your training_step when doing optimizations manually.
+        By using this we can ensure that all the proper scaling when using 16-bit etc has been done for you
+        
+        This function forwards all args to the .backward() call as well.
 
         Example::
 
-            def backward(self, trainer, loss, optimizer, optimizer_idx):
-                loss.backward()
+            def training_step(...):
+                loss = ...
 
-        """
-        loss.backward()
-
-    def amp_scale_loss(self, unscaled_loss: Tensor, optimizer: Optimizer, optimizer_idx: int, amp_backend: AMPType):
-        if amp_backend == AMPType.NATIVE:
-            scaled_loss = self.trainer.scaler.scale(unscaled_loss)
-        else:
-            scaled_loss = amp.scale_loss(unscaled_loss, optimizer)
+                # automatically applies scaling, etc...
+                self.backward(loss)
```

Comment:
While this is certainly an improvement, I'm just worried since one of our main principles was that "it's just organised Pytorch and no new framework to learn". I feel like this really tends towards not being really pytorchic (or how to call that). In PyTorch you call backward on tensors, not on models.

Reply:
welll... this is optional.

the user can always call loss.backward themselves.

But if they do, they don't get automatic mixed precision.

Reply:
```python
# manual option 1
loss = ...
loss.backward()
...

# option 2 (automates scaling, etc)
loss = ...
self.backward(loss, opt)
```